core/src/components/card/card.tsx

```diff
@@ -97,6 +102,20 @@ export class Card implements ComponentInterface, AnchorInterface, ButtonInterfac
     return this.href !== undefined || this.button;
   }
 
+  /**
+   * Disable the "xsmall" and "xlarge" sizes if the theme is "ios" or "md"
+   */
+  private getShape(): string | undefined {
```

Comment:
```suggestion
  private getShape() {
```

TypeScript is able to infer the return type here because `this.shape` is strongly typed. Adding `string | undefined` reduces the effectiveness of type checking because it allows you to compare `this.getShape()` to any string instead of the ones defined on the `shape` property.

Reply:
Removed function entirely: https://github.com/ionic-team/ionic-framework/pull/29292/commits/3d209fc24546829f16d9e78a52e55acb9d38768a